public/app/features/browse-dashboards/components/CreateNewButton.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+
+import { Button, ButtonGroup, Dropdown, Menu, MenuItem } from '@grafana/ui';
+import { t } from 'app/core/internationalization';
+
+interface Props {
+  /**
+   * Pass a folder UID in which the dashboard or folder will be created
+   */
+  inFolder?: string;
+}
+
+export function CreateNewButton({ inFolder }: Props) {
+  const newMenu = (
+    <Menu>
+      <MenuItem
+        url={addFolderUidToUrl('/dashboard/new', inFolder)}
+        label={t('search.dashboard-actions.new-dashboard', 'New Dashboard')}
+      />
+      <MenuItem
+        url={addFolderUidToUrl('/dashboards/folder/new', inFolder)}
+        label={t('search.dashboard-actions.new-folder', 'New Folder')}
+      />
+      <MenuItem
+        url={addFolderUidToUrl('/dashboard/import', inFolder)}
+        label={t('search.dashboard-actions.import', 'Import')}
+      />
+    </Menu>
+  );
+
+  return (
+    <Dropdown overlay={newMenu}>
+      <ButtonGroup>
+        <Button icon="plus">New</Button>
+        <Button icon="angle-down" />
+      </ButtonGroup>
+    </Dropdown>
+  );
+}
+
+/**
+ *
+ * @param url without any parameters
+ * @param folderUid  folder id
+ * @returns url with paramter if folder is present
+ */
+function addFolderUidToUrl(url: string, folderUid: string | undefined) {
```

Comment:
capitalize initialisms - `addFolderUIDToURL`

Reply:
In fact, I don't really think this function adds much value? Can just concat the string in the `url` prop  - i think that's fine

```
url={`/dashboard/import?folderUid=${inFolder}`}
```

Reply:
folderUid is undefined for the top level

Reply:
Missed that!!